# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp
# Commit: b003e26209cf
# Full Hash: b003e26209cf8e5dc1ac040f43f9caff01a5c613
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-04-09 04:12:49
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P2 PRemoteWorkerDebuggerMangaer binding implementation. r=asuth
#   
#   This patch implements the PRemoteWorkerDebuggerManager binding.
#   
#   PRemoteWorkerDebuggerManager is the IPC that handles the registration/unregistration of remote debuggers.
# ==============================================================================

diff -r 9e22fbf8ab8f -r b003e26209cf dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp
--- a/dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp	Tue Apr 08 09:43:33 2025 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerDebuggerManagerParent.cpp	Tue Apr 08 09:43:33 2025 +0000
@@ -8,6 +8,26 @@
 
 namespace mozilla::dom {
 
+/* static */
+RefPtr<RemoteWorkerDebuggerManagerParent>
+RemoteWorkerDebuggerManagerParent::CreateForProcess(
+    mozilla::ipc::Endpoint<PRemoteWorkerDebuggerManagerChild>* aChildEp) {
+  MOZ_ASSERT_DEBUG_OR_FUZZING(XRE_IsParentProcess() && NS_IsMainThread());
+
+  mozilla::ipc::Endpoint<PRemoteWorkerDebuggerManagerParent> parentEp;
+  nsresult rv =
+      PRemoteWorkerDebuggerManager::CreateEndpoints(&parentEp, aChildEp);
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return nullptr;
+  }
+
+  RefPtr<RemoteWorkerDebuggerManagerParent> actor =
+      MakeRefPtr<RemoteWorkerDebuggerManagerParent>();
+  parentEp.Bind(actor);
+
+  return actor;
+}
+
 RemoteWorkerDebuggerManagerParent::RemoteWorkerDebuggerManagerParent() {
   MOZ_ASSERT_DEBUG_OR_FUZZING(XRE_IsParentProcess() && NS_IsMainThread());
 }